### PR TITLE
Add extra error classes to handle server errors.

### PR DESCRIPTION
**Pull request by emancu**
In order to provide more granularity on the exceptions, we are adding some additional error classes for the common Server errors.

We also fixed the hierarchy for some client errors that were listed as server errors.

### CODE REVIEW

File: lib/json_api_client/errors.rb [hunk 1]
```diff
@@ -44,6 +44,28 @@ class AccessDenied < ClientError
     class NotAuthorized < ClientError
     end
 
+    class NotFound < ClientError
+      attr_reader :uri
+      def initialize(uri)
+        @uri = uri
+
+        msg = "Couldn't find resource at: #{uri.to_s}"
```
Comment (tilo) [inline lib/json_api_client/errors.rb:52]:
```suggestion
        msg = "Resource not found: #{uri.to_s}"
```

**Comment by gaorlov** [comment]
This is great! Can you please update the changelog md?
Thanks for taking the time!

**Comment by emancu** [comment]
@gaorlov done!

**Comment by gaorlov** [comment]
`1.19.0` is now live with your changes. Thanks for contributing!

**Pull request merged**